lsp/server/src/utils/orgUtils.ts

```diff
@@ -116,18 +76,24 @@ export class OrgUtils {
         }
     }
 
-    // Updates the auth state async
+    // Update the auth state async
     private static async checkAuthStatus(): Promise<AuthStatus> {
         if (this.authStatus !== AuthStatus.UNKNOWN) {
             return this.authStatus;
         }
+        try {
+            this.orgName = await this.getDefaultOrg();
+        } catch (e) {
+            this.orgName = undefined;
+        }
         const connection = await this.getConnection();
         if (connection === undefined) {
             //It is possible that orgName exists and connection expires
-            this.orgName = '';
```

Comment:
This line from previous version of the code is evil too. It's trying to do a service by if the connection is not available then set org's name to empty string. Which, you emulated but instead assign an undefined. Do you know if this line is necessary here or in your PR's case `this.orgName = undefined;`. Ideally `checkAuthStatus` itself shouldn't change the state but solely focus on doing what it claims to do. Principle of single responsiblity.

Reply:
`checkAuthStatus` is the central hub to maintain the state. The reason is listed in above comments.